doc/src/manual/conversion-and-promotion.md

```diff
@@ -118,15 +118,29 @@ its methods are restricted to cases that are considered "safe" or "unsurprising"
 (e.g. different representations of numbers, or different string encodings).
 It is also usually lossless; converting a value to a different type and back again
 should result in the exact same value.
-
 Notice that some constructors don't implement the concept of "conversion".
-For example, `Vector{Int}(5)` constructs a 5-element vector, which is not really a
-"conversion" from an integer to a vector.
+For example, `Timer(2)` creates a 2-second timer, which is not really a
+"conversion" from an integer to a timer.
 
-Finally, `convert(T, x)` is expected to return the original `x` if `x` is already of type `T`.
+`convert(T, x)` is expected to return the original `x` if `x` is already of type `T`.
 In contrast, if `T` is a mutable collection type then `T(x)` should always make a new
 collection (copying elements from `x`).
 
+The purpose of certain types is to "wrap" other values.
```

Comment:
At first it wasn't clear to me why you mentioned this. I think it would be useful to add a transition, like "For some types which "wrap" other values, the constructor may wrap its argument inside a new object even if it is already of the requested type."

More generally I find that the organization of this section is a bit messy. Maybe make it a bullet list, or add expressions to make the progression clearer? The part about `convert` not making copies could be moved to a different place (at the end of the section maybe?).